src/nix/search.md

```diff
@@ -62,10 +62,10 @@ R""(
 
 # Description
 
-`nix search` searches *installable* (which must be evaluatable, e.g. a
-flake) for packages whose name or description matches all of the
+`nix search` searches [*installable*](./nix.md#installables) (which can be evaluated, that is, a
+flake or Nix expression, but not a store path or store derivation path) for packages whose name or description matches all of the
```

Comment:
```suggestion
`nix search *evaluatable-installable* *regex*` searches its first argument (an [*evaluatable installable*](./nix.md#installables) for attributes whose values are installables whose name or description matches the regular expression *regex*.
```

Reply:
I agree this stuff needs to be more precise, and your suggestion is definitely an improvement. But this is not what this PR is about, so it would be great if you could make a follow-up.

Also I'm not sure why that aspect `nix search` behavior is even described here and not on its respective page. I also noticed this while rewording the description of the different installable types. Especially documentation for `--file` and `--expr`  should rather go into the option documentation, but this needs to be a different PR.